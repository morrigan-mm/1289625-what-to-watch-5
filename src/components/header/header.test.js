import React from "react";
import renderer from "react-test-renderer";
import {StaticRouter} from "react-router-dom";
import Header from "./header";

describe(`Header`, () => {
  it(`With no pageType`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <Header />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With pageType="movie-card"`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <Header pageType="movie-card" />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With pageType="user-page"`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <Header pageType="user-page" />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With content`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <Header>
              content
            </Header>
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
