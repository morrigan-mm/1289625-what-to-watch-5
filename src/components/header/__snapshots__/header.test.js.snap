// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header With content 1`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  content
</header>
`;

exports[`Header With no pageType 1`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</header>
`;

exports[`Header With pageType="movie-card" 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</header>
`;

exports[`Header With pageType="user-page" 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</header>
`;
